File: files/200-299/238/CH-238.R
library(tidyverse)
library(readxl)

path = "files/200-299/238/CH-238 Consecutive Numbers.xlsx"
input = read_excel(path, range = "B2:B19")
test = read_excel(path, range = "D2:E19")

result = input %>%
  mutate(group = cumsum(Numbers - lag(Numbers, default = first(Numbers)) != 1)) %>%
  mutate(Count = ifelse(n() == 1, "-", as.character(n())), .by = group) %>%
  select(Numbers, Count)

all.equal(result, test) # One row is different

File: files/200-299/238/CH-238.py
import pandas as pd

path = "200-299/238/CH-238 Consecutive Numbers.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=18)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=18).rename(columns=lambda col: col.replace('.1', ''))
test['Count'] = test['Count'].astype(str)

input.columns = ['Numbers']
g = (input['Numbers'] != input['Numbers'].shift(1) + 1).cumsum()
input['Count'] = input.groupby(g)['Numbers'].transform(lambda x: '-' if len(x)==1 else str(len(x)))
result = input[['Numbers', 'Count']]

print(result == test) # One row is different